base/utf16.jl

```diff
@@ -66,15 +76,190 @@ function encode16(s::AbstractString)
     UTF16String(buf)
 end
 
+function ascii_to_utf16(s::Array{UInt8,1})
+    len::Int = sizeof(s)
+    buf = Array{UInt16,1}(len+1)
+    for i = 1:len
```

Comment:
I couldn't find documentation for copy!, searching the docs at julialang only got me to copy,not copy!
```
copy(x)
Create a shallow copy of x: the outer structure is copied, but not all internal values. For example, copying an array produces a new array with identically-same elements as the original.
```

Does copy! widen from UInt8 to UInt16 by zero-extending then for you?
(It would be really nice if it used optimized code that did that 8 or more characters at a time,
like the code I'd written at my previous consulting gig...)